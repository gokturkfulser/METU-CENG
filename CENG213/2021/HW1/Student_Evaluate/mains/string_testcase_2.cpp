#include <iostream>

#include "String.h"
//#include "../String.h"

/*
 * Case 2 : Default constructor; various utility functions on empty string; print.
 */
int main() {
    std::cout << "-> Creating a string with default constructor." << std::endl;

    String str;

    std::cout << "-> Calculating the length." << std::endl;

    if (str.getLength() == 0) {
        std::cout << "length is zero" << std::endl;
    }

    std::cout << "-> Checking if the string is empty or not." << std::endl;

    if (str.isEmpty()) {
        std::cout << "string is empty" << std::endl;
    }

    std::cout << "-> Checking if the string is palindrome or not." << std::endl;

    if (str.isPalindrome()) {
        std::cout << "string is palindrome" << std::endl;
    }

    std::cout << "-> Printing the string." << std::endl;

    str.print(true);

    return 0;
}
