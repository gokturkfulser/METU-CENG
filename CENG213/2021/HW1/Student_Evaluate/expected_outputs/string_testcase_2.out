-> Creating a string with default constructor.
-> Calculating the length.
length is zero
-> Checking if the string is empty or not.
string is empty
-> Checking if the string is palindrome or not.
string is palindrome
-> Printing the string.
[]
